src/transaction/TransactionController.ts

```diff
@@ -691,26 +716,42 @@ export class TransactionController extends BaseController<
       throw new Error('No sign method defined.');
     }
 
-    const existingGasPrice = transactionMeta.transaction.gasPrice;
-    /* istanbul ignore next */
-    const existingGasPriceDecimal = parseInt(
-      existingGasPrice === undefined ? '0x0' : existingGasPrice,
-      16,
-    );
-    const gasPrice = addHexPrefix(
-      `${parseInt(`${existingGasPriceDecimal * CANCEL_RATE}`, 10).toString(
-        16,
-      )}`,
+    const gasPrice = getIncreasedPriceFromExisting(
+      transactionMeta.transaction.gasPrice,
+      CANCEL_RATE,
     );
 
-    const txParams = {
-      from: transactionMeta.transaction.from,
-      gasLimit: transactionMeta.transaction.gas,
-      gasPrice,
-      nonce: transactionMeta.transaction.nonce,
-      to: transactionMeta.transaction.from,
-      value: '0x0',
-    };
+    const existingMaxFeePerGas = transactionMeta.transaction?.maxFeePerGas;
+    const existingMaxPriorityFeePerGas =
+      transactionMeta.transaction?.maxPriorityFeePerGas;
+
+    const newMaxFeePerGas =
+      existingMaxFeePerGas &&
+      getIncreasedPriceFromExisting(existingMaxFeePerGas, CANCEL_RATE);
+    const newMaxPriorityFeePerGas =
+      existingMaxPriorityFeePerGas &&
+      getIncreasedPriceFromExisting(existingMaxPriorityFeePerGas, CANCEL_RATE);
+
+    const txParams =
+      newMaxFeePerGas && newMaxPriorityFeePerGas
+        ? {
+            from: transactionMeta.transaction.from,
+            gasLimit: transactionMeta.transaction.gas,
+            maxFeePerGas: newMaxFeePerGas,
+            maxPriorityFeePerGas: newMaxPriorityFeePerGas,
+            type: 2,
```

Comment:
Is this supposed to be in hex? Probably doesn't matter... Just looking at the ethereumjs/tx docs. Can't figure out why this field is necessary?

Reply:
you actually do need to specify `type: 2`: https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/tx/src/transactionFactory.ts#L22-L41 otherwise it assumes legacy